src/tests/ftest/soak/soak.py

```diff
@@ -0,0 +1,309 @@
+
+        """
+
+        # for the moment build IOR
+        #IorUtils.build_ior(self.basepath)
+
+        # read job info
+        job_params = "/run/" + job + "/"
+        job_name = self.params.get("name", job_params)
+        job_nodes = self.params.get("nodes", job_params)
+        job_processes = self.params.get("process_per_node",
+                                        job_params)
+        job_spec = self.params.get("jobspec", job_params)
+
+        # read ior cmd info
+        spec = "/run/" + job_spec + "/"
+        iteration = self.params.get("iter", spec + 'iteration/')
+        ior_flags = self.params.get("F", spec + 'iorflags/')
+        transfer_size = self.params.get("t", spec + 'transfersize/')
+        record_size = self.params.get("r", spec + 'recordsize/*')
+        stripe_size = self.params.get("s", spec + 'stripesize/*')
+        stripe_count = self.params.get("c", spec + 'stripecount/')
+        async_io = self.params.get("a", spec + 'asyncio/')
+        object_class = self.params.get("o", spec + 'objectclass/')
+
+        self.partition = self.params.get("partition",
+                                         '/run/hosts/test_machines/')
+
+        pool_uuid = self.pool.get_uuid_str()
+        tmplist = []
+        svc_list = ""
+        for i in range(self.createsvc):
+            tmplist.append(int(self.pool.svc.rl_ranks[i]))
+            svc_list += str(tmplist[i]) + ":"
+        svc_list = svc_list[:-1]
+
+        block_size = '1536m'
+
+        if stripe_size == '8m':
+            transfer_size = stripe_size
+
+        hostfile = os.path.join(self.tmpdir, "ior_hosts_" + job_name)
+
+        cmd = IorUtils.get_ior_cmd(ior_flags, iteration, block_size,
+                                   transfer_size, pool_uuid, svc_list,
+                                   record_size, stripe_size, stripe_count,
+                                   async_io, object_class, self.basepath,
+                                   hostfile, job_processes)
+
+        output = os.path.join(self.tmpdir, job_name + "_results.out")
+        script = slurm_utils.write_slurm_script(self.tmpdir, job_name,
+                                                output, int(job_nodes), [cmd])
+        return script
+
+    def setUp(self):
+
+        # intermediate results are stored in this global
+        # start off with it empty
+        self.soak_results = {}
+
+        self.partition = None
+
+        # initialize anything we rely on existing
+        self.pool = None
+        self.hostlist_servers = None
+
+        # get paths from the build_vars generated by build
+        with open('../../../.build_vars.json') as thefile:
+            build_paths = json.load(thefile)
+        self.basepath = os.path.normpath(build_paths['PREFIX']  + "/../")
+
+        # workdir was not successful, not sure why right now
+        self.tmpdir = self.basepath + "/install/tmp"
+        try:
+            os.makedirs(self.tmpdir)
+        except:
+            pass
```

Comment:
Why are we allowing this to `pass` on any exception?  If the intent is to `pass` if the dir already exists, then the `pass` exception should catch only that and any other exception should be fatal.